mitodl/micromasters
# refactor getPreferredName

#### What are the relevant tickets?

closes #1068 

#### What's this PR do?

This refactors the `getPreferredName` function to be a little smarter. Basically, it now has the correct precedence for first names (`preferred_name > first_name > username`).

#### Where should the reviewer start?

Read over the changes!

#### How should this be manually tested?

1. go to `/profile`
2. delete `preferred_name`, `first_name`, and `last_name` in the fields.
3. you should see `Hello ${username}`
4. add a first name
5. you should see `Hello ${first_name}`
6. add a last name
7. you should see `Hello ${first_name} ${last_name}`
8. add a preferred name
9. you should see `Hello ${preferred_name}`

I think that's it!

#### Any background context you want to provide?

#### Screenshots (if appropriate)

![preferred](https://cloud.githubusercontent.com/assets/6207644/18928164/f85a0486-858c-11e6-80fc-e064c797a263.png)

#### What GIF best describes this PR or how it makes you feel?
